Document how `TARGET_CC` (and not `CC`) must be set for Android (and other cross) targets

> cargo build --target aarch64-linux-android --release

failed to run custom build command for `ring v0.14.6`

Caused by:
  process didn't exit successfully: `/Users/bianca/IMPP/rust/target/release/build/ring-350e23b19999907d/build-script-build` (exit code: 101)
--- stdout
ANDROID_HOME: /Users/bianca/Library/Android/sdk
Apple_PubSub_Socket_Render: /private/tmp/com.apple.launchd.Sn7On4J2v6/Render
CARGO: /Users/bianca/.rustup/toolchains/stable-x86_64-apple-darwin/bin/cargo
CARGO_CFG_DEBUG_ASSERTIONS: 
CARGO_CFG_PROC_MACRO: 
CARGO_CFG_TARGET_ARCH: aarch64
CARGO_CFG_TARGET_ENDIAN: little
CARGO_CFG_TARGET_ENV: 
CARGO_CFG_TARGET_FAMILY: unix
CARGO_CFG_TARGET_OS: android
CARGO_CFG_TARGET_POINTER_WIDTH: 64
CARGO_CFG_TARGET_VENDOR: unknown
CARGO_CFG_UNIX: 
CARGO_FEATURE_DEFAULT: 1
CARGO_FEATURE_DEV_URANDOM_FALLBACK: 1
CARGO_FEATURE_USE_HEAP: 1
CARGO_HOME: /Users/bianca/.cargo
CARGO_MAKEFLAGS: --jobserver-fds=3,4 -j --jobserver-auth=3,4 -j
CARGO_MANIFEST_DIR: /Users/bianca/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6
CARGO_MANIFEST_LINKS: ring-asm
CARGO_PKG_AUTHORS: Brian Smith <brian@briansmith.org>
CARGO_PKG_DESCRIPTION: Safe, fast, small crypto using Rust.
CARGO_PKG_HOMEPAGE: 
CARGO_PKG_NAME: ring
CARGO_PKG_REPOSITORY: https://github.com/briansmith/ring
CARGO_PKG_VERSION: 0.14.6
CARGO_PKG_VERSION_MAJOR: 0
CARGO_PKG_VERSION_MINOR: 14
CARGO_PKG_VERSION_PATCH: 6
CARGO_PKG_VERSION_PRE: 
DEBUG: false
DYLD_FALLBACK_LIBRARY_PATH: /Users/bianca/IMPP/rust/target/release/deps:/Users/bianca/.rustup/toolchains/stable-x86_64-apple-darwin/lib:/Users/bianca/.rustup/toolchains/stable-x86_64-apple-darwin/lib:/Users/bianca/lib:/usr/local/lib:/usr/lib
HOME: /Users/bianca
HOST: x86_64-apple-darwin
LANG: de_DE.UTF-8
LOGNAME: bianca
NDK_HOME: /Users/bianca/Library/Android/sdk/ndk/20.0.5594570
NUM_JOBS: 4
OLDPWD: /Users/bianca/impp
OPT_LEVEL: 3
OUT_DIR: /Users/bianca/IMPP/rust/target/aarch64-linux-android/release/build/ring-173d20e891ec023d/out
PATH: /Users/bianca/.cargo/bin:/Users/bianca/.cargo/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/Users/bianca/Library/Android/sdk:/Users/bianca/Library/Android/sdk/tools:/Users/bianca/Library/Android/sdk/platform-tools
PROFILE: release
PWD: /Users/bianca/impp/rust
RUSTC: rustc
RUSTC_LINKER: /Users/bianca/.NDK/arm64/bin/aarch64-linux-android-clang
RUSTDOC: rustdoc
RUSTUP_HOME: /Users/bianca/.rustup
RUSTUP_TOOLCHAIN: stable-x86_64-apple-darwin
RUST_RECURSION_COUNT: 1
SHELL: /bin/bash
SHLVL: 1
SSH_AUTH_SOCK: /private/tmp/com.apple.launchd.gWOuAllj1C/Listeners
TARGET: aarch64-linux-android
TERM: xterm-256color
TERM_PROGRAM: Apple_Terminal
TERM_PROGRAM_VERSION: 421.2
TERM_SESSION_ID: FA2842F4-7A8C-4905-B7EC-6A4D8566A381
TMPDIR: /var/folders/mz/b2wg0jlj55b3jwt_44fshjx00000gn/T/
USER: bianca
XPC_FLAGS: 0x0
XPC_SERVICE_NAME: 0
_: /Users/bianca/.cargo/bin/cargo
__CF_USER_TEXT_ENCODING: 0x1F5:0x0:0x3
OPT_LEVEL = Some("3")
TARGET = Some("aarch64-linux-android")
HOST = Some("x86_64-apple-darwin")
CC_aarch64-linux-android = None
CC_aarch64_linux_android = None
TARGET_CC = None
CC = None
CFLAGS_aarch64-linux-android = None
CFLAGS_aarch64_linux_android = None
TARGET_CFLAGS = None
CFLAGS = None
CRATE_CC_NO_DEFAULTS = None
DEBUG = Some("false")
running "aarch64-linux-android-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=aarch64-linux-android" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/bianca/IMPP/rust/target/aarch64-linux-android/release/build/ring-173d20e891ec023d/out/aesv8-armx-linux64.o" "/Users/bianca/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/aesv8-armx-linux64.S"

--- stderr
thread 'main' panicked at 'failed to execute ["aarch64-linux-android-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=aarch64-linux-android" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/bianca/IMPP/rust/target/aarch64-linux-android/release/build/ring-173d20e891ec023d/out/aesv8-armx-linux64.o" "/Users/bianca/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/aesv8-armx-linux64.S"]: No such file or directory (os error 2)', /Users/bianca/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/build.rs:635:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

warning: build failed, waiting for other jobs to finish...

Similar issue on macOS Mojave. I'm cross-compiling and ring is a transitive dependancy:

`error: failed to run custom build command for `ring v0.14.6`

Caused by:
  process didn't exit successfully: `/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-1915a058b96cebea/build-script-build` (exit code: 101)
--- stdout
ANDROID_HOME: /Users/jeremy/Library/Android
ANDROID_NDK: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570
ANDROID_NDK_HOME: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570
ANDROID_NDK_TOOLCHAIN_DIR_PREBUILT: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin
ANDROID_PLATFORM_TOOLS: /Users/jeremy/Library/Android/sdk/platform-tools
ANDROID_SDK_HOME: /Users/jeremy/Library/Android/sdk
AR: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android-ar
Apple_PubSub_Socket_Render: /private/tmp/com.apple.launchd.HjaMde6Auh/Render
CARGO: /Users/jeremy/.rustup/toolchains/nightly-2019-05-22-x86_64-apple-darwin/bin/cargo
CARGO_CFG_DEBUG_ASSERTIONS: 
CARGO_CFG_PROC_MACRO: 
CARGO_CFG_TARGET_ARCH: x86_64
CARGO_CFG_TARGET_ENDIAN: little
CARGO_CFG_TARGET_ENV: 
CARGO_CFG_TARGET_FAMILY: unix
CARGO_CFG_TARGET_FEATURE: cmpxchg16b,fxsr,mmx,sse,sse2,sse3,ssse3
CARGO_CFG_TARGET_HAS_ATOMIC: 128,16,32,64,8,cas,ptr
CARGO_CFG_TARGET_OS: macos
CARGO_CFG_TARGET_POINTER_WIDTH: 64
CARGO_CFG_TARGET_THREAD_LOCAL: 
CARGO_CFG_TARGET_VENDOR: apple
CARGO_CFG_UNIX: 
CARGO_FEATURE_DEFAULT: 1
CARGO_FEATURE_DEV_URANDOM_FALLBACK: 1
CARGO_FEATURE_USE_HEAP: 1
CARGO_HOME: /Users/jeremy/.cargo
CARGO_MAKEFLAGS: --jobserver-fds=3,4 -j --jobserver-auth=3,4 -j
CARGO_MANIFEST_DIR: /Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6
CARGO_MANIFEST_LINKS: ring-asm
CARGO_PKG_AUTHORS: Brian Smith <brian@briansmith.org>
CARGO_PKG_DESCRIPTION: Safe, fast, small crypto using Rust.
CARGO_PKG_HOMEPAGE: 
CARGO_PKG_NAME: ring
CARGO_PKG_REPOSITORY: https://github.com/briansmith/ring
CARGO_PKG_VERSION: 0.14.6
CARGO_PKG_VERSION_MAJOR: 0
CARGO_PKG_VERSION_MINOR: 14
CARGO_PKG_VERSION_PATCH: 6
CARGO_PKG_VERSION_PRE: 
CC: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang
CFLAGS: 
CPPFLAGS: /usr/local/opt/openssl@1.1/include
CXX: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang++
DEBUG: false
DYLD_FALLBACK_LIBRARY_PATH: /Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/deps:/Users/jeremy/.rustup/toolchains/nightly-2019-05-22-x86_64-apple-darwin/lib:/Users/jeremy/.rustup/toolchains/nightly-2019-05-22-x86_64-apple-darwin/lib:/Users/jeremy/lib:/usr/local/lib:/usr/lib
HOME: /Users/jeremy
HOST: x86_64-apple-darwin
LC_CTYPE: UTF-8
LDFLAGS: 
LESS: -R
LOGNAME: jeremy
LSCOLORS: Gxfxcxdxbxegedabagacad
NUM_JOBS: 8
OLDPWD: /Users/jeremy/kzen/pingpong-wallet/pingpong-react/android
OPT_LEVEL: 3
OUT_DIR: /Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out
PAGER: less
PATH: /Users/jeremy/.cargo/bin:/usr/local/opt/openssl@1.1/bin:/Users/jeremy/.cargo/bin:/Users/jeremy/.cargo/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/jeremy/Library/Android/sdk/platform-tools:/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin:/Users/jeremy/Library/Android/sdk/platform-tools:/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin
PKG_CONFIG_ALLOW_CROSS: 1
PROFILE: release
PWD: /Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust
RUSTC: rustc
RUSTC_LINKER: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang
RUSTDOC: rustdoc
RUSTUP_HOME: /Users/jeremy/.rustup
RUSTUP_TOOLCHAIN: nightly-2019-05-22-x86_64-apple-darwin
RUST_RECURSION_COUNT: 1
SECURITYSESSIONID: 186a9
SHELL: /bin/zsh
SHLVL: 2
SSH_AUTH_SOCK: /private/tmp/com.apple.launchd.xvKKZVDAU4/Listeners
SYSROOT: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin//../sysroot
TARGET: x86_64-apple-darwin
TERM: xterm-256color
TERM_PROGRAM: Apple_Terminal
TERM_PROGRAM_VERSION: 421.2
TERM_SESSION_ID: 334DC91F-41BA-4497-A11F-AFB7EC528FBE
TMPDIR: /var/folders/32/whl0d9wd0m70xr2r38g2yxlw0000gn/T/
USER: jeremy
XPC_FLAGS: 0x0
XPC_SERVICE_NAME: 0
ZSH: /Users/jeremy/.oh-my-zsh
_: /Users/jeremy/.cargo/bin/cargo
__CF_USER_TEXT_ENCODING: 0x1F5:0x0:0x0
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = Some("")
DEBUG = Some("false")
running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/aes-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/aes-x86_64-macosx.S"
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = Some("")
DEBUG = Some("false")
running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/aesni-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/aesni-x86_64-macosx.S"
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = Some("")
DEBUG = Some("false")
running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/vpaes-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/vpaes-x86_64-macosx.S"
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = Some("")
DEBUG = Some("false")
running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/x86_64-mont-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/x86_64-mont-macosx.S"
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = Some("")
DEBUG = Some("false")
running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/x86_64-mont5-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/x86_64-mont5-macosx.S"
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = Some("")
DEBUG = Some("false")
running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/chacha-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/chacha-x86_64-macosx.S"
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = Some("")
DEBUG = Some("false")
running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/p256-x86_64-asm-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/p256-x86_64-asm-macosx.S"
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = Some("")
DEBUG = Some("false")
running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/aesni-gcm-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/aesni-gcm-x86_64-macosx.S"
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = Some("")
DEBUG = Some("false")
running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/ghash-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/ghash-x86_64-macosx.S"
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = Some("")
DEBUG = Some("false")
running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/poly1305-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/poly1305-x86_64-macosx.S"
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = Some("")
DEBUG = Some("false")
running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/sha512-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/sha512-x86_64-macosx.S"
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = Some("")
DEBUG = Some("false")
running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/sha256-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/sha256-x86_64-macosx.S"
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = Some("")
DEBUG = Some("false")
running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-std=c1x" "-Wbad-function-cast" "-Wmissing-prototypes" "-Wnested-externs" "-Wstrict-prototypes" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/block.o" "crypto/block.c"

--- stderr
In file included from crypto/block.c:15:
In file included from crypto/internal.h:112:
include/GFp/base.h:65:10: fatal error: 'sys/types.h' file not found
#include <sys/types.h>
         ^~~~~~~~~~~~~
1 error generated.
thread 'main' panicked at 'execution failed', /Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/build.rs:638:9
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

warning: build failed, waiting for other jobs to finish...
error: build failed`


Cargo config:

`[target.aarch64-linux-android]
ar = "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android-ar"
linker = "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang"

[target.armv7-linux-androideabi]
ar = "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/arm-linux-androideabi-ar"
linker = "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/armv7a-linux-androideabi23-clang"

[target.i686-linux-android]
ar = "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/i686-linux-android-ar"
linker = "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/i686-linux-android23-clang"`


Any help is much appreciated.

> Similar issue on macOS Mojave. I'm cross-compiling and ring is a transitive dependancy:
> 
> `error: failed to run custom build command for `ring v0.14.6`
> 
> Caused by:
> process didn't exit successfully: `/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-1915a058b96cebea/build-script-build` (exit code: 101)
> --- stdout
> ANDROID_HOME: /Users/jeremy/Library/Android
> ANDROID_NDK: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570
> ANDROID_NDK_HOME: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570
> ANDROID_NDK_TOOLCHAIN_DIR_PREBUILT: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin
> ANDROID_PLATFORM_TOOLS: /Users/jeremy/Library/Android/sdk/platform-tools
> ANDROID_SDK_HOME: /Users/jeremy/Library/Android/sdk
> AR: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android-ar
> Apple_PubSub_Socket_Render: /private/tmp/com.apple.launchd.HjaMde6Auh/Render
> CARGO: /Users/jeremy/.rustup/toolchains/nightly-2019-05-22-x86_64-apple-darwin/bin/cargo
> CARGO_CFG_DEBUG_ASSERTIONS:
> CARGO_CFG_PROC_MACRO:
> CARGO_CFG_TARGET_ARCH: x86_64
> CARGO_CFG_TARGET_ENDIAN: little
> CARGO_CFG_TARGET_ENV:
> CARGO_CFG_TARGET_FAMILY: unix
> CARGO_CFG_TARGET_FEATURE: cmpxchg16b,fxsr,mmx,sse,sse2,sse3,ssse3
> CARGO_CFG_TARGET_HAS_ATOMIC: 128,16,32,64,8,cas,ptr
> CARGO_CFG_TARGET_OS: macos
> CARGO_CFG_TARGET_POINTER_WIDTH: 64
> CARGO_CFG_TARGET_THREAD_LOCAL:
> CARGO_CFG_TARGET_VENDOR: apple
> CARGO_CFG_UNIX:
> CARGO_FEATURE_DEFAULT: 1
> CARGO_FEATURE_DEV_URANDOM_FALLBACK: 1
> CARGO_FEATURE_USE_HEAP: 1
> CARGO_HOME: /Users/jeremy/.cargo
> CARGO_MAKEFLAGS: --jobserver-fds=3,4 -j --jobserver-auth=3,4 -j
> CARGO_MANIFEST_DIR: /Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6
> CARGO_MANIFEST_LINKS: ring-asm
> CARGO_PKG_AUTHORS: Brian Smith [brian@briansmith.org](mailto:brian@briansmith.org)
> CARGO_PKG_DESCRIPTION: Safe, fast, small crypto using Rust.
> CARGO_PKG_HOMEPAGE:
> CARGO_PKG_NAME: ring
> CARGO_PKG_REPOSITORY: https://github.com/briansmith/ring
> CARGO_PKG_VERSION: 0.14.6
> CARGO_PKG_VERSION_MAJOR: 0
> CARGO_PKG_VERSION_MINOR: 14
> CARGO_PKG_VERSION_PATCH: 6
> CARGO_PKG_VERSION_PRE:
> CC: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang
> CFLAGS:
> CPPFLAGS: /usr/local/opt/openssl@1.1/include
> CXX: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang++
> DEBUG: false
> DYLD_FALLBACK_LIBRARY_PATH: /Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/deps:/Users/jeremy/.rustup/toolchains/nightly-2019-05-22-x86_64-apple-darwin/lib:/Users/jeremy/.rustup/toolchains/nightly-2019-05-22-x86_64-apple-darwin/lib:/Users/jeremy/lib:/usr/local/lib:/usr/lib
> HOME: /Users/jeremy
> HOST: x86_64-apple-darwin
> LC_CTYPE: UTF-8
> LDFLAGS:
> LESS: -R
> LOGNAME: jeremy
> LSCOLORS: Gxfxcxdxbxegedabagacad
> NUM_JOBS: 8
> OLDPWD: /Users/jeremy/kzen/pingpong-wallet/pingpong-react/android
> OPT_LEVEL: 3
> OUT_DIR: /Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out
> PAGER: less
> PATH: /Users/jeremy/.cargo/bin:/usr/local/opt/openssl@1.1/bin:/Users/jeremy/.cargo/bin:/Users/jeremy/.cargo/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/jeremy/Library/Android/sdk/platform-tools:/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin:/Users/jeremy/Library/Android/sdk/platform-tools:/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin
> PKG_CONFIG_ALLOW_CROSS: 1
> PROFILE: release
> PWD: /Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust
> RUSTC: rustc
> RUSTC_LINKER: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang
> RUSTDOC: rustdoc
> RUSTUP_HOME: /Users/jeremy/.rustup
> RUSTUP_TOOLCHAIN: nightly-2019-05-22-x86_64-apple-darwin
> RUST_RECURSION_COUNT: 1
> SECURITYSESSIONID: 186a9
> SHELL: /bin/zsh
> SHLVL: 2
> SSH_AUTH_SOCK: /private/tmp/com.apple.launchd.xvKKZVDAU4/Listeners
> SYSROOT: /Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin//../sysroot
> TARGET: x86_64-apple-darwin
> TERM: xterm-256color
> TERM_PROGRAM: Apple_Terminal
> TERM_PROGRAM_VERSION: 421.2
> TERM_SESSION_ID: 334DC91F-41BA-4497-A11F-AFB7EC528FBE
> TMPDIR: /var/folders/32/whl0d9wd0m70xr2r38g2yxlw0000gn/T/
> USER: jeremy
> XPC_FLAGS: 0x0
> XPC_SERVICE_NAME: 0
> ZSH: /Users/jeremy/.oh-my-zsh
> _: /Users/jeremy/.cargo/bin/cargo
> __CF_USER_TEXT_ENCODING: 0x1F5:0x0:0x0
> OPT_LEVEL = Some("3")
> TARGET = Some("x86_64-apple-darwin")
> HOST = Some("x86_64-apple-darwin")
> CC_x86_64-apple-darwin = None
> CC_x86_64_apple_darwin = None
> HOST_CC = None
> CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
> CFLAGS_x86_64-apple-darwin = None
> CFLAGS_x86_64_apple_darwin = None
> HOST_CFLAGS = None
> CFLAGS = Some("")
> DEBUG = Some("false")
> running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/aes-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/aes-x86_64-macosx.S"
> OPT_LEVEL = Some("3")
> TARGET = Some("x86_64-apple-darwin")
> HOST = Some("x86_64-apple-darwin")
> CC_x86_64-apple-darwin = None
> CC_x86_64_apple_darwin = None
> HOST_CC = None
> CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
> CFLAGS_x86_64-apple-darwin = None
> CFLAGS_x86_64_apple_darwin = None
> HOST_CFLAGS = None
> CFLAGS = Some("")
> DEBUG = Some("false")
> running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/aesni-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/aesni-x86_64-macosx.S"
> OPT_LEVEL = Some("3")
> TARGET = Some("x86_64-apple-darwin")
> HOST = Some("x86_64-apple-darwin")
> CC_x86_64-apple-darwin = None
> CC_x86_64_apple_darwin = None
> HOST_CC = None
> CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
> CFLAGS_x86_64-apple-darwin = None
> CFLAGS_x86_64_apple_darwin = None
> HOST_CFLAGS = None
> CFLAGS = Some("")
> DEBUG = Some("false")
> running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/vpaes-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/vpaes-x86_64-macosx.S"
> OPT_LEVEL = Some("3")
> TARGET = Some("x86_64-apple-darwin")
> HOST = Some("x86_64-apple-darwin")
> CC_x86_64-apple-darwin = None
> CC_x86_64_apple_darwin = None
> HOST_CC = None
> CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
> CFLAGS_x86_64-apple-darwin = None
> CFLAGS_x86_64_apple_darwin = None
> HOST_CFLAGS = None
> CFLAGS = Some("")
> DEBUG = Some("false")
> running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/x86_64-mont-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/x86_64-mont-macosx.S"
> OPT_LEVEL = Some("3")
> TARGET = Some("x86_64-apple-darwin")
> HOST = Some("x86_64-apple-darwin")
> CC_x86_64-apple-darwin = None
> CC_x86_64_apple_darwin = None
> HOST_CC = None
> CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
> CFLAGS_x86_64-apple-darwin = None
> CFLAGS_x86_64_apple_darwin = None
> HOST_CFLAGS = None
> CFLAGS = Some("")
> DEBUG = Some("false")
> running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/x86_64-mont5-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/x86_64-mont5-macosx.S"
> OPT_LEVEL = Some("3")
> TARGET = Some("x86_64-apple-darwin")
> HOST = Some("x86_64-apple-darwin")
> CC_x86_64-apple-darwin = None
> CC_x86_64_apple_darwin = None
> HOST_CC = None
> CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
> CFLAGS_x86_64-apple-darwin = None
> CFLAGS_x86_64_apple_darwin = None
> HOST_CFLAGS = None
> CFLAGS = Some("")
> DEBUG = Some("false")
> running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/chacha-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/chacha-x86_64-macosx.S"
> OPT_LEVEL = Some("3")
> TARGET = Some("x86_64-apple-darwin")
> HOST = Some("x86_64-apple-darwin")
> CC_x86_64-apple-darwin = None
> CC_x86_64_apple_darwin = None
> HOST_CC = None
> CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
> CFLAGS_x86_64-apple-darwin = None
> CFLAGS_x86_64_apple_darwin = None
> HOST_CFLAGS = None
> CFLAGS = Some("")
> DEBUG = Some("false")
> running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/p256-x86_64-asm-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/p256-x86_64-asm-macosx.S"
> OPT_LEVEL = Some("3")
> TARGET = Some("x86_64-apple-darwin")
> HOST = Some("x86_64-apple-darwin")
> CC_x86_64-apple-darwin = None
> CC_x86_64_apple_darwin = None
> HOST_CC = None
> CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
> CFLAGS_x86_64-apple-darwin = None
> CFLAGS_x86_64_apple_darwin = None
> HOST_CFLAGS = None
> CFLAGS = Some("")
> DEBUG = Some("false")
> running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/aesni-gcm-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/aesni-gcm-x86_64-macosx.S"
> OPT_LEVEL = Some("3")
> TARGET = Some("x86_64-apple-darwin")
> HOST = Some("x86_64-apple-darwin")
> CC_x86_64-apple-darwin = None
> CC_x86_64_apple_darwin = None
> HOST_CC = None
> CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
> CFLAGS_x86_64-apple-darwin = None
> CFLAGS_x86_64_apple_darwin = None
> HOST_CFLAGS = None
> CFLAGS = Some("")
> DEBUG = Some("false")
> running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/ghash-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/ghash-x86_64-macosx.S"
> OPT_LEVEL = Some("3")
> TARGET = Some("x86_64-apple-darwin")
> HOST = Some("x86_64-apple-darwin")
> CC_x86_64-apple-darwin = None
> CC_x86_64_apple_darwin = None
> HOST_CC = None
> CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
> CFLAGS_x86_64-apple-darwin = None
> CFLAGS_x86_64_apple_darwin = None
> HOST_CFLAGS = None
> CFLAGS = Some("")
> DEBUG = Some("false")
> running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/poly1305-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/poly1305-x86_64-macosx.S"
> OPT_LEVEL = Some("3")
> TARGET = Some("x86_64-apple-darwin")
> HOST = Some("x86_64-apple-darwin")
> CC_x86_64-apple-darwin = None
> CC_x86_64_apple_darwin = None
> HOST_CC = None
> CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
> CFLAGS_x86_64-apple-darwin = None
> CFLAGS_x86_64_apple_darwin = None
> HOST_CFLAGS = None
> CFLAGS = Some("")
> DEBUG = Some("false")
> running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/sha512-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/sha512-x86_64-macosx.S"
> OPT_LEVEL = Some("3")
> TARGET = Some("x86_64-apple-darwin")
> HOST = Some("x86_64-apple-darwin")
> CC_x86_64-apple-darwin = None
> CC_x86_64_apple_darwin = None
> HOST_CC = None
> CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
> CFLAGS_x86_64-apple-darwin = None
> CFLAGS_x86_64_apple_darwin = None
> HOST_CFLAGS = None
> CFLAGS = Some("")
> DEBUG = Some("false")
> running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/sha256-x86_64-macosx.o" "/Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/pregenerated/sha256-x86_64-macosx.S"
> OPT_LEVEL = Some("3")
> TARGET = Some("x86_64-apple-darwin")
> HOST = Some("x86_64-apple-darwin")
> CC_x86_64-apple-darwin = None
> CC_x86_64_apple_darwin = None
> HOST_CC = None
> CC = Some("/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang")
> CFLAGS_x86_64-apple-darwin = None
> CFLAGS_x86_64_apple_darwin = None
> HOST_CFLAGS = None
> CFLAGS = Some("")
> DEBUG = Some("false")
> running "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-I" "include" "-std=c1x" "-Wbad-function-cast" "-Wmissing-prototypes" "-Wnested-externs" "-Wstrict-prototypes" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-D_XOPEN_SOURCE=700" "-c" "-o/Users/jeremy/kzen/pingpong-wallet/pingpong-react/android/rust/target/release/build/ring-c7757965f2b77b28/out/block.o" "crypto/block.c"
> 
> --- stderr
> In file included from crypto/block.c:15:
> In file included from crypto/internal.h:112:
> include/GFp/base.h:65:10: fatal error: 'sys/types.h' file not found
> #include <sys/types.h>
> ^~~~~~~~~~~~~
> 1 error generated.
> thread 'main' panicked at 'execution failed', /Users/jeremy/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/build.rs:638:9
> note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
> 
> warning: build failed, waiting for other jobs to finish...
> error: build failed`
> 
> Cargo config:
> 
> `[target.aarch64-linux-android]
> ar = "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android-ar"
> linker = "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android23-clang"
> 
> [target.armv7-linux-androideabi]
> ar = "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/arm-linux-androideabi-ar"
> linker = "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/armv7a-linux-androideabi23-clang"
> 
> [target.i686-linux-android]
> ar = "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/i686-linux-android-ar"
> linker = "/Users/jeremy/Library/Android/sdk/ndk/20.0.5594570/toolchains/llvm/prebuilt/darwin-x86_64/bin/i686-linux-android23-clang"`
> 
> Any help is much appreciated.

I switched to a different crate (webpage instead of ureq) which doesn't has ring as a dependency, maybe that's an option for you as well.

--- a/BUILDING.md
+++ b/BUILDING.md
@@ -33,7 +33,19 @@ use Perl as a macro assembly language.
 When building from Git for Windows, directories containing yasm.exe and
 perl.exe must be in `%PATH%`, where yasm.exe is
 [Yasm](http://yasm.tortall.net/Download.html) 1.3 or later and where perl.exe
-is recommended to be [Strawberry Perl](http://strawberryperl.com). 
+is recommended to be [Strawberry Perl](http://strawberryperl.com).
+
+
+Cross Compiling
+---------------
+
+When you build *ring* for a target that is different than the one you are using
+for the build process you need to install the rust tool chain and a C/C++
+compiler that can produce binaries for the intended target.
+
+Besides the required dependencies you need to set the environment variables
+`TARGET_CC` and `TARGET_AR` to the full path of the cross-compiler and the
+cross-archiver respectively.
 
 
 Supported Toolchains

--- a/README.md
+++ b/README.md
@@ -50,8 +50,9 @@ See the documentation at
 https://briansmith.org/rustdoc/ring/.
 
 See [BUILDING.md](BUILDING.md) for instructions on how to build it. These
-instructions are especially important on Windows when not building from
-crates.io, as there are build prerequisites that need to be installed.
+instructions are especially important for cross-compiling and for building on
+Windows when not building from crates.io, as there are build prerequisites that
+need to be installed.
 
 
 

